src/eliminate_data_type.cpp

```diff
@@ -42,31 +108,36 @@ void eliminate_data_type::apply(module& m) const
                                                            "scatternd_add",
                                                            "scatternd_mul",
                                                            "scatternd_none"};
-    for(auto ins : iterator_for(m))
+    if(unsupported_types.empty() and unsupported_fp8_ops.empty())
+    {
+        return;
+    }
+    else if(not unsupported_fp8_ops.empty() and not unsupported_types.empty())
+    {
+        MIGRAPHX_THROW("eliminate_data_type: specify either unsupported FP8 ops or unsupported "
+                       "data types not both.");
+    }
+    else if(unsupported_fp8_ops.empty())
+    {
+        for(auto ins : iterator_for(m))
+        {
+            if(ins->name()[0] == '@')
+                continue;
+            if(contains(skip_op_names, ins->name()))
+                continue;
+            insert_convert_to_supported_type(m, ins, target_type, unsupported_types);
+        }
+    }
+    else
     {
-        if(ins->name()[0] == '@')
-            continue;
-        if(contains(skip_op_names, ins->name()))
-            continue;
-        auto inputs = ins->inputs();
-        std::transform(inputs.begin(), inputs.end(), inputs.begin(), [&](auto i) {
-            if(types.count(i->get_shape().type()) == 0)
-                return i;
-            return m.insert_instruction(ins, make_op("convert", {{"target_type", target_type}}), i);
-        });
-        if(inputs == ins->inputs())
-            continue;
-        auto op         = ins->get_operator();
-        auto attributes = op.attributes();
-        if(attributes.contains("general_data_type"))
+        std::set<migraphx::shape::type_t> unsupported_fp8_types = {
```

Comment:
Fp8 should not be hardcoded here. Instead it should be passed through the `types` variable.

Reply:
Done.